StRoot/StETofUtil/StETofGeometry.cxx

```diff
@@ -1488,4 +1572,82 @@ StETofGeometry::getFieldZ( const double& x, const double& y, const double& z ) {
     mStarBField->BField( X, B );
 
     return B[ 2 ];
-}
\ No newline at end of file
+}
+
+void 
+StETofGeometry::readAlignmentParameters(){
+      LOG_INFO << "etofAlignParam: filename provided --> use parameter file: " << mFileNameAlignParam.c_str() << endm;
+        
+		  //add check for no alignment set here.
+	
+		  ifstream paramFile;		
+        paramFile.open( mFileNameAlignParam.c_str() );
+
+        if( !paramFile.is_open() ) {
+            LOG_INFO << "unable to get the alignments parameters from file --> file does not exist" << endm;
+            return;
+        }
+
+	     StThreeVectorD counterAlignmentParameter;
+        float tempX;
+		  float tempY;
+		  float tempZ;
+        while( paramFile >> tempX >> tempY >> tempZ ) {
+		  counterAlignmentParameter = StThreeVectorD( tempX, tempY, tempZ );
+            mAlignmentParameters.push_back( counterAlignmentParameter );
+        }
+        
+        paramFile.close();
+
+        if( mAlignmentParameters.size() != 108 ) {
+            LOG_INFO << "parameter file for alignments has not the right amount of entries: ";
+            LOG_INFO << mAlignmentParameters.size() << " instead of 108 !!!!" << endm;
+            return;
+        }
+
+}
+
+void 
+StETofGeometry::readAlignmentDatabase(StMaker* maker){
+        LOG_INFO << "No etof alignment file provided --> use database: " << endm;
+        
+		  //add check for no alignment set here.
+	
+    	  TDataSet* dbDataSet = nullptr;
+
+		 if(!maker){
+			LOG_WARN << "No valid pointer to any StMaker given. Do not load alignment from database" << endm;
+			return;
+		 }
+
+        dbDataSet = maker->GetDataBase( "Geometry / etof / etofAlign" );
+
+        St_etofAlign* etofAlign = static_cast< St_etofAlign * > ( dbDataSet->Find( "etofAlign" ) );
+        if( !etofAlign ) {
+            LOG_ERROR << "unable to get the align params from the database" << endm;
+            return;
+        }
+
+        etofAlign_st* table = etofAlign->GetTable();
+	     StThreeVectorD counterAlignmentParameter;
+        float tempX;
+		  float tempY;
+		  float tempZ;
+
+		  for( int iCounter = 0; iCounter < eTofConst::nCounters; iCounter++){
+       		tempX = table[iCounter].detectorAlignX[0];
+       		tempY = table[iCounter].detectorAlignY[0];
+       		tempZ = table[iCounter].detectorAlignZ[0];
+		  		StThreeVectorD counterAlignmentParameter = StThreeVectorD( tempX, tempY, tempZ );
+            mAlignmentParameters.push_back( counterAlignmentParameter );
+        }
+
+        if( mAlignmentParameters.size() != 108 ) {
+            LOG_WARN << "parameter vector for alignments has not the right amount of entries: ";
+            LOG_WARN << mAlignmentParameters.size() << " instead of 108 !!!!" << endm;
+            return;
+        }else{
+				return;
+		 }
```

Comment:
mAlignmentParameters is a private member, you just filled it so, why check the vector size? Don't you trust yourself? Also, does eTofConst::nCounters equal 108? Why use the literal then?